model/execution/DeliveryExecutionList.php

```diff
@@ -0,0 +1,336 @@
+    {
+        $now = microtime(true);
+        $remaining = $this->getRemainingTime($cachedData);
+        $elapsedApprox = 0;
+        $executionState = $cachedData[DeliveryMonitoringService::STATUS];
+
+        if (
+            $executionState === DeliveryExecution::STATE_ACTIVE
+            && isset($cachedData[DeliveryMonitoringService::LAST_TEST_TAKER_ACTIVITY])
+        ) {
+            $lastActivity = $cachedData[DeliveryMonitoringService::LAST_TEST_TAKER_ACTIVITY];
+            $elapsedApprox = $now - $lastActivity;
+            $duration = isset($cachedData[DeliveryMonitoringService::ITEM_DURATION])
+                ? (float)$cachedData[DeliveryMonitoringService::ITEM_DURATION]
+                : 0;
+            $elapsedApprox += $duration;
+        }
+
+        if (is_bool($online) && $online === false) {
+            $elapsedApprox = 0;
+        }
+
+        return round((float)$remaining - $elapsedApprox);
+    }
+
+    /**
+     * @param array $cachedData
+     * @return int
+     */
+    private function getRemainingTime(array $cachedData)
+    {
+        $remaining = isset($cachedData[DeliveryMonitoringService::REMAINING_TIME])
+            ? (int)$cachedData[DeliveryMonitoringService::REMAINING_TIME]
+            : 0;
+        return $remaining;
+    }
+
+
+    /**
+     * Get array of user specific extra fields to be displayed in the monitoring data table
+     *
+     * @return array
+     * @throws \common_ext_ExtensionException
+     */
+    private function getUserExtraFields()
+    {
+        $extraFields = [];
+        $proctoringExtension = $this->getExtensionManagerService()->getExtensionById('taoProctoring');
+        $userExtraFields = $proctoringExtension->getConfig('monitoringUserExtraFields');
+        if (empty($userExtraFields) || !is_array($userExtraFields)) {
+            return [];
+        }
+
+        $userExtraFieldsSettings = $proctoringExtension->getConfig('monitoringUserExtraFieldsSettings');
+        foreach ($userExtraFields as $name => $uri) {
+            $extraFields[] = $this->mergeExtraFieldsSettings($uri, $name, $userExtraFieldsSettings);
+        }
+
+        return $extraFields;
+    }
+
+    private function mergeExtraFieldsSettings($uri, $name, $userExtraFieldsSettings)
+    {
+        $property = $this->getProperty($uri);
+        $settings = array_key_exists($name, $userExtraFieldsSettings)
+            ? $userExtraFieldsSettings[$name]
+            : [];
+
+        return array_merge([
+            'id' => $name,
+            'property' => $property,
+            'label' => __($property->getLabel()),
+        ], $settings);
+    }
+
+    /**
+     * @param array $cachedData
+     * @return mixed|string
+     */
+    private function getProgressString(array $cachedData)
+    {
+        $progressStr = $cachedData[DeliveryMonitoringService::CURRENT_ASSESSMENT_ITEM];
+
+        if (($progress = json_decode($progressStr, true)) === null) {
```

Comment:
Please split this in two lines, for the sake of reading. YOu already have the json decoded into a variable, no need to put it on the same line:
```
$progress = json_decode($progressStr, true);
if ($progress === null) {
```